orchagent/bfdorch.cpp

```diff
@@ -0,0 +1,412 @@
+#include "bfdorch.h"
+#include "converter.h"
+#include "swssnet.h"
+#include "notifier.h"
+#include "sai_serialize.h"
+#include "directory.h"
+
+using namespace std;
+using namespace swss;
+
+#define BFD_SESSION_DEFAULT_TX_INTERVAL 1000
+#define BFD_SESSION_DEFAULT_RX_INTERVAL 1000
+#define BFD_SESSION_DEFAULT_DETECT_MULTIPLIER 3
+
+extern sai_bfd_api_t*       sai_bfd_api;
+extern sai_object_id_t      gSwitchId;
+extern sai_object_id_t      gVirtualRouterId;
+extern PortsOrch*           gPortsOrch;
+extern Directory<Orch*>     gDirectory;
+
+const map<string, sai_bfd_session_type_t> session_type_map =
+{
+    {"demand_active",       SAI_BFD_SESSION_TYPE_DEMAND_ACTIVE},
+    {"demand_passive",      SAI_BFD_SESSION_TYPE_DEMAND_PASSIVE},
+    {"async_active",        SAI_BFD_SESSION_TYPE_ASYNC_ACTIVE},
+    {"async_passive",       SAI_BFD_SESSION_TYPE_ASYNC_PASSIVE}
+};
+
+const map<sai_bfd_session_type_t, string> session_type_loopup =
+{
+    {SAI_BFD_SESSION_TYPE_DEMAND_ACTIVE,    "demand_active"},
+    {SAI_BFD_SESSION_TYPE_DEMAND_PASSIVE,   "demand_passive"},
+    {SAI_BFD_SESSION_TYPE_ASYNC_ACTIVE,     "async_active"},
+    {SAI_BFD_SESSION_TYPE_ASYNC_PASSIVE,    "async_passive"}
+};
+
+const map<sai_bfd_session_state_t, string> session_state_loopup =
+{
+    {SAI_BFD_SESSION_STATE_ADMIN_DOWN,  "Admin_Down"},
+    {SAI_BFD_SESSION_STATE_DOWN,        "Down"},
+    {SAI_BFD_SESSION_STATE_INIT,        "Init"},
+    {SAI_BFD_SESSION_STATE_UP,          "Up"}
+};
+
+BfdOrch::BfdOrch(DBConnector *db, string tableName):
+    Orch(db, tableName)
+{
+    SWSS_LOG_ENTER();
+
+    m_state_db = shared_ptr<DBConnector>(new DBConnector("STATE_DB", 0));
+    m_stateBfdSessionTable = unique_ptr<Table>(new Table(m_state_db.get(), STATE_BFD_SESSION_TABLE));
```

Comment:
As mentioned, please pass this attributes from orch init

Reply:
Thanks for the comment. Moved the attributes to orch init.